pkg/generate/code/resource_reference.go

```diff
@@ -99,27 +125,47 @@ func ReferenceFieldsPresent(
 	crd *model.CRD,
 	sourceVarName string,
 ) string {
-	out := "false"
+	iteratorsOut := ""
+	returnOut := "return false"
 	fieldAccessPrefix := fmt.Sprintf("%s%s", sourceVarName,
 		crd.Config().PrefixConfig.SpecField)
 	// Sorted fieldnames are used for consistent code-generation
-	for _, fieldName := range crd.SortedFieldNames() {
+	for fieldIndex, fieldName := range crd.SortedFieldNames() {
 		field := crd.Fields[fieldName]
 		if field.HasReference() {
-			out += " || ("
 			fp := fieldpath.FromString(field.Path)
 			// remove fieldName from fieldPath before adding nil checks
 			// for nested fieldPath
 			fp.Pop()
+
+			// Determine whether the field is nested
+			if fp.Size() > 0 {
+				// Determine whether the field is inside a slice
+				parentField, ok := crd.Fields[fp.String()]
+				if !ok {
+					panic(fmt.Sprintf("CRD %s has no Field with path %s", crd.Kind, fp.String()))
+				}
+
+				if parentField.ShapeRef.Shape.Type == "list" {
+					iteratorsOut += fmt.Sprintf("for _, iter%d := range %s.%s {\n", fieldIndex, fieldAccessPrefix, parentField.Path)
```

Comment:
This should be preceded by a nil check or it will throw nil pointer dereference error inside range `r.ko.Spec.FieldA.FieldB.ListFieldC`

There is a generator method already written to generate nil check for nested fields.

Reply:
Yep. Not necessary for this particularly example I'm testing against, since `range ko.Spec.Routes` will just iterate 0 times in `Routes` is `nil`. But easy enough to add